File: VB/WizardCustomizationQueryPage/CustomChooseConnectionPageView.vb
Imports DevExpress.DataAccess
Imports DevExpress.DataAccess.Native
Imports DevExpress.DataAccess.Sql
Imports DevExpress.DataAccess.UI.Native
Imports DevExpress.DataAccess.UI.Wizard.Views
Imports DevExpress.DataAccess.Wizard.Views
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace WizardCustomizationQueryPage
    Partial Public Class CustomChooseConnectionPageView
        Inherits WizardViewBase
        Implements IChooseConnectionPageView

        Private lbConnections As DevExpress.XtraEditors.ListBoxControl

        Private anyExceptions_Renamed As Boolean = False

        Public Sub New(ByVal connections As IEnumerable(Of SqlDataConnection))
            InitializeComponent()
            AddHandler lbConnections.SelectedIndexChanged, AddressOf listBoxControl1_SelectedIndexChanged
            Me.lbConnections.Items.Clear()
            For Each dataConnection As SqlDataConnection In connections
                If TypeOf dataConnection Is DataConnection Then
                    Me.lbConnections.Items.Add(New ConnectionListBoxItem(dataConnection))
                End If
            Next dataConnection
        End Sub

        Private Sub listBoxControl1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
            RaiseEvent Changed(Me, New EventArgs())
        End Sub

        Public Overrides ReadOnly Property HeaderDescription() As String
            Get
                Return "Choose one connection from the list."
            End Get
        End Property
        Public Event Changed As EventHandler Implements IChooseConnectionPageView.Changed

        Public ReadOnly Property ExistingConnectionName() As String Implements IChooseConnectionPageView.ExistingConnectionName
            Get
                Return (DirectCast(CType(lbConnections.SelectedItem, ConnectionListBoxItem).Connection, INamedItem)).Name
            End Get
        End Property

        Public Sub SetSelectedConnection(ByVal connection As INamedItem) Implements IChooseConnectionPageView.SetSelectedConnection
            If connection IsNot Nothing Then
                For Each connectionListBoxItem As ConnectionListBoxItem In Me.lbConnections.Items
                    If DirectCast(connectionListBoxItem.Connection, INamedItem).Name = connection.Name Then
                        lbConnections.SelectedItem = connectionListBoxItem
                        Exit For
                    End If
                Next connectionListBoxItem
            End If
        End Sub

        Public ReadOnly Property ShouldCreateNewConnection() As Boolean Implements IChooseConnectionPageView.ShouldCreateNewConnection
            Get
                Return False
            End Get
        End Property

        Public ReadOnly Property WaitFormActivator() As IWaitFormActivator Implements IChooseConnectionPageView.WaitFormActivator
            Get
                Return New WaitFormActivator(FindForm(), GetType(WaitFormWithCancel), True)
            End Get
        End Property

        Public ReadOnly Property AnyExceptions() As Boolean Implements IChooseConnectionPageView.AnyExceptions
            Get
                Return anyExceptions_Renamed
            End Get
        End Property

        Public Sub HandleException(ByVal exception As Exception) Implements IChooseConnectionPageView.HandleException
            MessageBox.Show(exception.Message)
        End Sub
    End Class
    Public Class ConnectionListBoxItem
        Private privateConnection As SqlDataConnection
        Public Property Connection() As SqlDataConnection
            Get
                Return privateConnection
            End Get
            Private Set(ByVal value As SqlDataConnection)
                privateConnection = value
            End Set
        End Property
        Public Sub New(ByVal connection As SqlDataConnection)
            Me.Connection = connection
        End Sub
        Public Overrides Function ToString() As String
            Return Connection.Name
        End Function
    End Class
End Namespace

File: CS/WizardCustomizationQueryPage/CustomChooseConnectionPageView.cs
using DevExpress.DataAccess;
using DevExpress.DataAccess.Native;
using DevExpress.DataAccess.Sql;
using DevExpress.DataAccess.UI.Native;
using DevExpress.DataAccess.UI.Wizard.Views;
using DevExpress.DataAccess.Wizard.Views;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WizardCustomizationQueryPage {
    public partial class CustomChooseConnectionPageView : WizardViewBase, IChooseConnectionPageView {
        private DevExpress.XtraEditors.ListBoxControl lbConnections;
        bool anyExceptions = false;

        public CustomChooseConnectionPageView(IEnumerable<SqlDataConnection> connections) {
            InitializeComponent();
            lbConnections.SelectedIndexChanged += listBoxControl1_SelectedIndexChanged;
            this.lbConnections.Items.Clear();
            foreach(SqlDataConnection dataConnection in connections) {
                if(dataConnection is DataConnection)
                    this.lbConnections.Items.Add(new ConnectionListBoxItem(dataConnection));
            }
        }

        void listBoxControl1_SelectedIndexChanged(object sender, EventArgs e) {
            if(Changed != null)
                Changed(this, new EventArgs());
        }

        public override string HeaderDescription {
            get { return "Choose one connection from the list."; }
        }
        public event EventHandler Changed;

        public string ExistingConnectionName {
            get { return ((INamedItem)((ConnectionListBoxItem)lbConnections.SelectedItem).Connection).Name; }
        }

        public void SetSelectedConnection(INamedItem connection) {
            if(connection != null)
                foreach(ConnectionListBoxItem connectionListBoxItem in this.lbConnections.Items) {
                    if(((INamedItem)connectionListBoxItem.Connection).Name == connection.Name) {
                        lbConnections.SelectedItem = connectionListBoxItem;
                        break;
                    }
                }
        }

        public bool ShouldCreateNewConnection { get { return false; } }

        public IWaitFormActivator WaitFormActivator {
            get { return new WaitFormActivator(FindForm(), typeof(WaitFormWithCancel), true); }
        }

        public bool AnyExceptions { get { return anyExceptions; } }

        public void HandleException(Exception exception) {
            MessageBox.Show(exception.Message);
        }
    }
    public class ConnectionListBoxItem { 
        public SqlDataConnection Connection { get; private set; }
        public ConnectionListBoxItem(SqlDataConnection connection) {
            Connection = connection;
        }
        public override string ToString()
        {
            return Connection.Name;
        }
    }
}
